src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -65,6 +64,7 @@ internal enum PreconditionState
             if (fileLength.HasValue)
             {
                 SetAcceptRangeHeader(context);
+                // Sets the Content-Length header to the file length. This is overwritten for valid range requests with the length of the range.
```

Comment:
> Sets the Content-Length header to the file length.

This is totally obvious, comments should the things you can't understand from a trivial reading of the code 😆 

> This is overwritten for valid range requests with the length of the range.

I would instead say something like: "Right now we're just setting the content length to the whole file, which means we're assuming this isn't a range request. If this is a range request, the code that processes the range header will change the content length to the size of the range."

You want to explain what the non-obvious things and assumptions are 👍